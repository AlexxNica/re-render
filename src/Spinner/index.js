import React from 'react';
import { ActivityIndicator } from 'react-native';

export default class Spinner extends React.Component {
    render() {
        return (
            <ActivityIndicator color={this.props.color}
                               size={this.props.size}
                               animating={true}
                               hidesWhenStopped={true}/>
        )
    }
}

Spinner.propTypes = {
    color: React.PropTypes.string,
    size: React.PropTypes.number,
};