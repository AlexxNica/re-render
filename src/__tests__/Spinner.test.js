import React from 'react';
import renderer from 'react-test-renderer';
import Spinner from '../Spinner/index';

it('renders correctly', () => {
    const tree = renderer.create(
        <Spinner color={'red'} size={20}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
