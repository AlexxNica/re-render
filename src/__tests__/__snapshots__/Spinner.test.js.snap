// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  className="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-height-z80fyv rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-width-19wmn03"
  role="progressbar"
>
  <div
    className="rn-alignItems-1oszu61 rn-animationDuration-17bb2tj rn-animationIterationCount-1muvv40 rn-animationName-dozj4v rn-animationTimingFunction-1ldzwu0 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        strokeWidth="4"
        style={
          Object {
            "opacity": 0.2,
            "stroke": "red",
          }
        }
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        strokeWidth="4"
        style={
          Object {
            "stroke": "red",
            "strokeDasharray": 80,
            "strokeDashoffset": 60,
          }
        }
      />
    </svg>
  </div>
</div>
`;
